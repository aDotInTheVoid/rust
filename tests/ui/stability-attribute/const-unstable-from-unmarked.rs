// aux-build: const-unstable.rs
// compile-flags: -Zforce-unstable-if-unmarked
#![crate_type = "lib"]
extern crate const_unstable;

// Check that crates build with `-Zforce-unstable-if-unmarked` can't call
// const-unstable functions, despite their functions sometimes being considerd
// unstable.
//
// See https://github.com/rust-lang/rust/pull/118427#discussion_r1409914941 for
// more context.

pub const fn identity(x: i32) -> i32 {
    const_unstable::identity(x)
    //~^ ERROR `const_unstable::identity` is not yet stable as a const fn
}
