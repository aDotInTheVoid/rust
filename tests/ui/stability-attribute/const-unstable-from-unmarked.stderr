error: `const_unstable::identity` is not yet stable as a const fn
  --> $DIR/const-unstable-from-unmarked.rs:14:5
   |
LL |     const_unstable::identity(x)
   |     ^^^^^^^^^^^^^^^^^^^^^^^^^^^
   |
   = help: add `#![feature(const_identity)]` to the crate attributes to enable

error: aborting due to 1 previous error

